src/core/namespace.c

```diff
@@ -995,6 +995,13 @@ static int mount_private_dev(MountEntry *m) {
         if (r < 0)
                 log_debug_errno(r, "Failed to set up basic device tree at '%s', ignoring: %m", temporary_mount);
 
+        /* Remount read-only. */
+        r = mount_nofollow_verbose(LOG_DEBUG, NULL, dev, NULL,
+                                   MS_REMOUNT|DEV_MOUNT_OPTIONS|MS_RDONLY,
+                                   "mode=0755" TMPFS_LIMITS_PRIVATE_DEV);
```

Comment:
this would make the super block read only. Which would mean you are making the whole of /tmp/ read-only.

We should rather make the bind mount read-only. i.e. MS_REMOUNT|MS_BIND|MS_RDONLY, and a NULL options pointer.